coding-articles/Article.md

```diff
@@ -0,0 +1,584 @@
+*/
+};
+```
+The second purpose is to use the network manager object to initialize the provider object for the network connection. (If you don't know what a network provider check [this](https://docs.ethers.io/v5/api/providers/#providers)).
+
+To do so, we need to install [forta-agent-tools](https://www.npmjs.com/package/forta-agent-tools)
+
+```
+npm i forta-agent-tools
+```
+
+Go to `src/agent.ts` 
+
+**Delete all** content inside it and import the `NetworkManager` class from the installed package
+
+```
+import { NetworkManager } from "forta-agent-tools";
+```
+
+Create the object with the `networkData`. I created on the [network.ts](./code/src/network.ts)
+
+```
+// below the import NetworkManager
+import { networkData } from "./network";
+const networkManager = new NetworkManager(networkData); //global var
+```
+
+To add the provider, we need to call `networkManager.init(<provider>)` , but we need this to be called only one time when the bot is **initialized**. Forta has a built-in function to [initialize](https://docs.forta.network/en/latest/sdk/#handlers)
+
+```
+import ethers from "forta-agent"
+// below the import and the networkManager object
+export const provideInitialize = (
+  provider: ethers.providers.JsonRpcProvider
+) => {
+   // should return a function that will be used by the Bot
+  return async () => {
+    await networkManager.init(provider);
+  };
+};
+```
+
+The last thing export the provider initialize
+```
+export default {
+  initialize: provideInitialize(utils.provider), // The utils object will be added on the next section
+};
+```
+Now in `src/agent.ts` you should have something like this
+```
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData, networkData } from "./network";
+const networkManager = new NetworkManager(networkData);
+export const provideInitialize = (
+  provider: ethers.providers.JsonRpcProvider
+) => {
+   // should return a function that will be used by the Bot
+  return async () => {
+    await networkManager.init(provider);
+  };
+};
+export default {
+  initialize: provideInitialize(utils.provider), // The utils object will be added on the next section
+};
+```
+
+Inside the bot, we have the network manager and all data that we need. The only missing thing is the `handleTransaction` and the `utils` object. I will start with the utils object. I will create it in a separate file
+
+```
+nano src/utils.ts
+```
+Add the provider
+
+```
+import { getEthersProvider } from "forta-agent";
+const provider = getEthersProvider();
+export default {
+  provider
+};
+```
+Because we are already on the utils.ts let's add everything like the finding reports and the event ABI.
+The event ABI, We need the event abi to be able to catch it when it triggers
+
+```
+// below getEthersProvider import
+import { Interface } from "@ethersproject/abi";
+const EVENT_ABI: string[] = ["event Deposited(uint256 amount)"];
+const EVENTS_IFACE: Interface = new Interface(EVENT_ABI);
+export default {
```

Comment:
I think you missed exporting '_createFinding_' here.

Reply:
truee